rfcs/0101-nix-formatting.md

```diff
@@ -858,28 +899,102 @@ inherit
 
 **Description**
 
-Barring some exceptions, on multiline items, the closing semicolon is on its own line and without indentation.
+- Usually the semicolon is placed directly at the end of the binder. There are exceptions in which the semicolon is placed onto the following line instead:
+  - Multiline `inherit` statements
+  - Binders in which the value is a multiline `if` expression or nested operator.
 
 **Examples**
 
 ```nix
-foo = bar;
-foo = function call {
-  # stuff
-};
-foo =
-  let
-   foo = "bar"
-  in
-  some statement
-;
+{
+  attr1 = bar;
+  attr2 = function call {
+    # stuff
+  };
+  attr3 =
+    function call
+      many
+      arguments;
+  attr3 =
+    let
+    foo = "bar"
+    in
+    some statement;
+  attr4 =
+    if foo then
+      "bar"
+    else
+      "baz"
+  ;
+  attr5 =
+    let
+      foo = false;
+    in
+    if foo then "bar" else "baz";
+  attr6 = function (
+    if foo then
+      "bar"
+    else
+      "baz"
+  );
+  attr7 =
+    cond1
+    || cond2
+    ||
+      some function call
+      && cond3
+  ;
+}
 ```
 
 **Rationale and alternatives**
 
 There are three possible locations to put the semicolon:
-Always directly at the end of the content, always on a new line with one indentation level, always on a new lien without indentation.
-
-On a new line with indentation is out, as it looks inferior to without indentation: Without indentation, the semicolon acts as a closing token which visually ends the statement.
+1. Directly at the end of the content.
+  - This may result in weird placements, especially when interacting with `if` expressions.
```

Comment:
I don't think that's a valid argument, "weirdness" is very subjective. An argument that does hold though is that this style makes changes harder (can't just append to the last line) and it makes diffs larger (e.g. adding a final `else if` clause changes one additional line).